Source/Core/AST/Absy.cs

```diff
@@ -2167,10 +2177,17 @@ public override void Emit(TokenTextWriter stream, int level)
         stream.WriteLine();
         stream.WriteLine("}");
       }
-      else
+      else if (!this.DefinitionAxioms.Any())
       {
         stream.WriteLine(";");
       }
+      if (this.DefinitionAxioms.Any())
```

Comment:
Could you move 

```
if (this.DefinitionAxioms.Any())
      {
        stream.WriteLine();
        stream.WriteLine("uses {");
        this.DefinitionAxioms.ForEach(axiom => axiom.Emit(stream, level));
        stream.WriteLine("}");
      }
```

as part of the chained if-then-else construction above? Then you won't need the change on line 2180.

Reply:
Would that work when a function has both a body and definition axioms? I think with the `:inline` attribute this becomes possible.

Example:
```
function {:inline} foo(x: int): bool {
    // some body
} uses {
    // some axioms
}
```

Reply:
ok